code/modules/reagents/reagent_dispenser.dm

```diff
@@ -190,82 +210,107 @@
+	icon             = 'icons/obj/objects.dmi'
+	icon_state       = "peppertank"
+	anchored         = TRUE
+	density          = FALSE
+	amount_dispensed = 45
+
+/obj/structure/reagent_dispensers/peppertank/populate_reagents()
+	reagents.add_reagent(/decl/material/liquid/capsaicin/condensed, reagents.maximum_volume)
 
 /obj/structure/reagent_dispensers/water_cooler
-	name = "water cooler"
-	desc = "A machine that dispenses cool water to drink."
-	amount_per_transfer_from_this = 5
-	icon = 'icons/obj/vending.dmi'
-	icon_state = "water_cooler"
+	name                      = "water cooler"
+	desc                      = "A machine that dispenses cool water to drink."
+	icon                      = 'icons/obj/vending.dmi'
+	icon_state                = "water_cooler"
 	possible_transfer_amounts = null
-	anchored = 1
-	initial_capacity = 500
-	initial_reagent_types = list(/decl/material/liquid/water = 1)
-	tool_interaction_flags = (TOOL_INTERACTION_ANCHOR | TOOL_INTERACTION_DECONSTRUCT)
-	var/cups = 12
-	var/cup_type = /obj/item/chems/drinks/sillycup
+	amount_dispensed          = 5
+	anchored                  = TRUE
+	volume                    = 500
+	tool_interaction_flags    = (TOOL_INTERACTION_ANCHOR | TOOL_INTERACTION_DECONSTRUCT)
+	var/cups                  = 12
+	var/tmp/max_cups          = 12
+	var/tmp/cup_type          = /obj/item/chems/drinks/sillycup
+
+/obj/structure/reagent_dispensers/water_cooler/populate_reagents()
+	reagents.add_reagent(/decl/material/liquid/water, reagents.maximum_volume)
 
 /obj/structure/reagent_dispensers/water_cooler/attack_hand(var/mob/user)
+	return dispense_cup(user)
+
+/obj/structure/reagent_dispensers/water_cooler/proc/dispense_cup(var/mob/user, var/skip_text = FALSE)
 	if(cups > 0)
-		var/visible_messages = DispenserMessages(user)
-		visible_message(visible_messages[1], visible_messages[2])
 		var/cup = new cup_type(loc)
 		user.put_in_active_hand(cup)
 		cups--
-	else
-		to_chat(user, RejectionMessage(user))
-
-/obj/structure/reagent_dispensers/water_cooler/proc/DispenserMessages(var/mob/user)
-	return list("\The [user] grabs a paper cup from \the [src].", "You grab a paper cup from \the [src]'s cup compartment.")
+		if(!skip_text)
+			visible_message("\The [user] grabs a paper cup from \the [src].", "You grab a paper cup from \the [src]'s cup compartment.")
+		return TRUE
 
-/obj/structure/reagent_dispensers/water_cooler/proc/RejectionMessage(var/mob/user)
-	return "The [src]'s cup dispenser is empty."
+	if(!skip_text)
+		to_chat(user, "The [src]'s cup dispenser is empty.")
 
 /obj/structure/reagent_dispensers/water_cooler/attackby(obj/item/W, mob/user)
+	//Allow refilling with a box
+	if((cups < max_cups) && istype(W, /obj/item/storage))
+		var/obj/item/storage/S = W
+		for(var/obj/item/chems/drinks/C in S)
+			if(cups >= max_cups)
+				break
+			if(istype(C, cup_type))
+				S.remove_from_storage(C, src)
+				qdel(C)
+				cups++
+		return TRUE
+	
 	. = ..()
-	if(!.)
+	if(!. && ATOM_IS_OPEN_CONTAINER(W))
```

Comment:
Because otherwise the animation would play on any unhandled interactions.

Reply:
We don't wanna return when the parent returns false, because reagent_containers interactions are handled in after_attack, not in attackby itself:
```

/obj/item/chems/glass/afterattack(var/obj/target, var/mob/user, var/proximity)
	if(!ATOM_IS_OPEN_CONTAINER(src) || !proximity) //Is the container open & are they next to whatever they're clicking?
		return FALSE //If not, do nothing.
	for(var/type in can_be_placed_into) //Is it something it can be placed into?
		if(istype(target, type))
			return TRUE
	if(standard_dispenser_refill(user, target)) //Are they clicking a water tank/some dispenser?
		return TRUE
	if(standard_pour_into(user, target)) //Pouring into another beaker?
		return TRUE
	if(standard_feed_mob(user, target))
		return TRUE
	if(user.a_intent == I_HURT)
		if(standard_splash_mob(user,target))
			return TRUE
		if(reagents && reagents.total_volume)
			to_chat(user, SPAN_DANGER("You splash the contents of \the [src] onto \the [target]."))
			reagents.splash(target, reagents.total_volume)
			return TRUE
	else if(reagents && reagents.total_volume)
		to_chat(user, SPAN_NOTICE("You splash a small amount of the contents of \the [src] onto \the [target]."))
		reagents.splash(target, min(reagents.total_volume, 5))
		return TRUE
	. = ..()
```

Reply:
No problems!